daemon/oci_linux.go

```diff
@@ -751,37 +768,56 @@ func (daemon *Daemon) createSpec(c *container.Container) (retSpec *specs.Spec, e
 	}
 
 	if err := daemon.initCgroupsPath(parentPath); err != nil {
-		return nil, fmt.Errorf("linux init cgroups path: %v", err)
+		// all of the errors in initCgroupsPath are things like failing to make
+		// a directory, or are otherwise the result of a function call with
+		// hard-coded arguments, so return a System error if it fails.
+		return nil, errdefs.System(fmt.Errorf("linux init cgroups path: %v", err))
 	}
 	if err := setDevices(&s, c); err != nil {
-		return nil, fmt.Errorf("linux runtime spec devices: %v", err)
+		// The errors resulting from devices are generally more of a
+		// user-configured-it-wrong error, so return InvalidParameter
+		return nil, errdefs.InvalidParameter(fmt.Errorf("linux runtime spec devices: %v", err))
 	}
 	if err := daemon.setRlimits(&s, c); err != nil {
-		return nil, fmt.Errorf("linux runtime spec rlimits: %v", err)
+		// as of this writing, setRlimits doesn't return any errors at all, so
+		// we return error type Unknown if it happens.
+		return nil, errdefs.Unknown(fmt.Errorf("linux runtime spec rlimits: %v", err))
```

Comment:
`errors.Wrap`
But we should maybe not use errdefs here, in case `setRLimits` wants to use errors int he future? Or if it's really not returning errors, maybe the method definition should be updated.

Reply:
I could do either. I suppose it would be the ideal to enforce that if `setRLimits` ever did return errors, they would have to be of `errdefs` types.